code/modules/modular_computers/file_system/program.dm

```diff
@@ -83,32 +83,43 @@
 // Check if the user can run program. Only humans can operate computer. Automatically called in run_program()
 // User has to wear their ID or have it inhand for ID Scan to work.
 // Can also be called manually, with optional parameter being access_to_check to scan the user's ID
-/datum/computer_file/program/proc/can_run(var/mob/living/user, var/loud = 0, var/access_to_check)
+/datum/computer_file/program/proc/can_run(var/mob/living/user, var/loud = 0, var/list/accesses_to_check, var/datum/computer_network/network)
 	if(!requires_access_to_run)
-		return 1
-	// Defaults to required_access
-	if(!access_to_check)
-		access_to_check = required_access
-	if(!access_to_check) // No required_access, allow it.
-		return 1
+		return TRUE
+	// Defaults to network accecss, and then required_access
+	if(!length(accesses_to_check))
+		if(network)
+			var/datum/extension/network_device/acl/access_controller = network.access_controller
+			if(access_controller)
+				accesses_to_check = access_controller.get_program_access(filename)
+				if(!length(accesses_to_check))
+					accesses_to_check = list(required_access)
+				else if(accesses_to_check[1] == "NONE")
```

Comment:
Really not a fan of dummy NONE access like this. Empty list should signify this. If you need to inherit default access some but not all of the time, add a bool and/or proc to the program itself and check/call that to get extra default accesses with a different inheritance structure.